packages/cdk-graph/src/filtering/presets.ts

```diff
@@ -0,0 +1,145 @@
+/*********************************************************************************************************************
+ Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License").
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ******************************************************************************************************************** */
+import { ConstructOrder } from "constructs";
+import { FlagEnum, Graph } from "../core";
+import { IGraphFilterPlan } from "./types";
+
+/**
+ * Verify that store is filterable, meaning it allows destructive mutations.
+ * @throws Error if store is not filterable
+ * @internal
+ */
+export function verifyFilterable(store: Graph.Store): void {
+  if (!store.allowDestructiveMutations) {
+    throw new Error(
+      "Store must allow destructive mutations to perform filtering; clone store before applying filters"
```

Comment:
be more explicit and give the clone method to call.

Reply:
Acknowledged